### PR TITLE
SMTChecker: Remove UnderOverflow verification target for BMC engine

### PR DESCRIPTION
**Pull request by pgebal**
Refactor. Use Overflow and Underflow verification targets instead of OverUnderflow verification target introduced earlier for convenience. That simplifies code in BMC engine.

### CODE REVIEW

File: test/libsolidity/smtCheckerTests/control_flow/branches_with_return/constructor_state_variable_init.sol [hunk 1]
```diff
@@ -40,3 +40,4 @@ contract C is B {
 // Warning 6328: (489-513): CHC: Assertion violation happens here.
 // Warning 6328: (533-546): CHC: Assertion violation happens here.
 // Info 1391: CHC: 3 verification condition(s) proved safe! Enable the model checker option "show proved safe" to see all of them.
+// Info 6002: BMC: 1 verification condition(s) proved safe! Enable the model checker option "show proved safe" to see all of them.
```
Comment (blishko) [inline test/libsolidity/smtCheckerTests/control_flow/branches_with_return/constructor_state_variable_init.sol:43]: Why do we have a new BMC report here about verification target proved safe?
Is it this unary minus? `constructor(int a) A(-a)`?
It looks like we are missing some verification target in CHC?

Comment (pgebal) [reply test/libsolidity/smtCheckerTests/control_flow/branches_with_return/constructor_state_variable_init.sol:43]: yes, it's unary minus

Comment (pgebal) [reply test/libsolidity/smtCheckerTests/control_flow/branches_with_return/constructor_state_variable_init.sol:43]: It looks like we don't have it for CHC.
In BMC it's handled in `BMC::endVisit(UnaryOperation const& _op)`
In CHC `CHC::endVisit(UnaryOperation const& _op)` we handle only user defined functions.

**Review by blishko — APPROVED** [review]
LGMT!

**Pull request merged**